### PR TITLE
APIGOV-22882 - resource instance prior to converting to Access Request

### PR DESCRIPTION
**Pull request by jcollins-axway**

### CODE REVIEW

File: pkg/agent/cache/accessrequest.go [hunk 1]
```diff
@@ -44,11 +45,13 @@ func (c *cacheManager) GetAccessRequestByAppAndAPI(appName, remoteAPIID, remoteA
...
-	accessRequest, _ := c.accessRequestMap.GetBySecondaryKey(appName + ":" + remoteAPIID + ":" + remoteAPIStage)
-	if accessRequest != nil {
-		ri, ok := accessRequest.(*mv1.AccessRequest)
-		if ok {
-			return ri
+	cachedAccessRequest, _ := c.accessRequestMap.GetBySecondaryKey(appName + ":" + remoteAPIID + ":" + remoteAPIStage)
+	if cachedAccessRequest != nil {
+		if ri, ok := cachedAccessRequest.(*v1.ResourceInstance); ok {
+			accessRequest := &mv1.AccessRequest{}
+			if accessRequest.FromInstance(ri) == nil {
```
Comment (sbolosan) [inline pkg/agent/cache/accessrequest.go:52]: If it's nil you wanna return? Or != nil? Slightly confused. No code in front of me.

 Still 🏃🏃🏃

Comment (jcollins-axway) [reply pkg/agent/cache/accessrequest.go:52]: FromInstance returns an error only. The object that calls it is loaded to from the resource instance that was sent in.

Comment (sbolosan) [reply pkg/agent/cache/accessrequest.go:52]: 👍🏾

**Review by sbolosan — APPROVED** [review]
Good call on changing to using resource

**Review by vivekschauhan — APPROVED** [review]

**Pull request merged**